monty_profiling/floppy/flop_counting/operations/numpy_functions.py

```diff
@@ -0,0 +1,291 @@
+import numpy as np
+from typing import Any
+
+__all__ = [
+    "SumOperation",
+    "ClipOperation",
+    "WhereOperation",
+    "MinOperation",
+    "MaxOperation",
+    "RoundOperation",
+    "IsnanOperation",
+    "ArgminOperation",
+    "ArgmaxOperation",
+    "TraceOperation",
+    "MeanOperation",
+    "StdOperation",
+    "VarOperation",
+    "AverageOperation",
+    "LogOperation",
+    "PowerOperation",
+    "FloorDivideOperation",
+    "ModuloOperation",
+    "BitwiseAndOperation",
+    "BitwiseOrOperation",
+]
+class SumOperation:
+    """FLOP count for sum operation."""
+    def count_flops(self, *args: Any, result: Any) -> int:
+        """Count FLOPs for sum operation."""
+        return np.size(args[0]) - 1
```

Comment:
I think this will depend if we're just summing a vector/summing along the first dimension, vs. summing along multiple dimensions?

Reply:
Yeah good point about upper-bounding, maybe just add a comment to the doc-string to make that clear, i.e. we ignore reduce from summing along specific axes.